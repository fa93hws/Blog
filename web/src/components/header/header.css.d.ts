// This file is automatically generated.
// Please do not change this file!
interface CssExports {
  'colorBlue': string;
  'colorPurpleLight': string;
  'container': string;
  'content': string;
  'cross': string;
  'gridBaseline': string;
  'monoBlackA250': string;
  'monoWhite': string;
  'monoWhiteA200': string;
  'monoWhiteA50': string;
  'shortBar': string;
  'subTitle': string;
  'withBottomShadow': string;
}
export const cssExports: CssExports;
export default cssExports;
