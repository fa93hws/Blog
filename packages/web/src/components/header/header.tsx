import * as React from 'react';
import { AppBar, Toolbar, Typography } from '@material-ui/core';

export const Header = React.memo(() => (
  <AppBar position="static">
    <Toolbar>
      <Typography variant="h6">夏目天子的博客</Typography>
    </Toolbar>
  </AppBar>
));
