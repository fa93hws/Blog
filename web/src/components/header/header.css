@value gridBaseline from '../metrics.css';
@value colorBlue from '../colors.css';
@value monoWhite, monoWhiteA50, monoWhiteA200, monoBlackA250 from '../colors.css';
@value colorPurpleLight from '../colors.css';

.cross {
  fill: monoWhite;
  background-color: unset;
  border: none;
  cursor: pointer;
  margin-left: calc(gridBaseline * 3);
}

.container {
  background-color: colorBlue;
  color: monoWhiteA50;
  height: calc(gridBaseline * 30);
}

.shortBar {
  position: fixed;
  top: 0;
  height: calc(gridBaseline * 7);
  width: 100%;
  display: flex;
  align-items: center;
}

.withBottomShadow {
  box-shadow: 0 0 gridBaseline 0 monoBlackA250;
}

.content {
  padding-left: calc(gridBaseline * 30);
  padding-top: calc(gridBaseline * 13);
}

.subTitle {
  color: colorPurpleLight;
}
