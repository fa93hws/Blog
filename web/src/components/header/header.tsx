import * as React from 'react';
import * as classnames from 'classnames';
import styles from './header.css';
import { Title, Text } from '../typography/typography';
import { Cross } from '../icons/icons';

export const Header = ({
  showShortBarTitle,
  onCloseClicked,
}: {
  showShortBarTitle: boolean;
  onCloseClicked(): void;
}) => (
  <div className={styles.container}>
    <div
      className={classnames(styles.shortBar, {
        [styles.withBottomShadow]: showShortBarTitle,
      })}
    >
      <button className={styles.cross} onClick={onCloseClicked}>
        <Cross size="small" title="close" />
      </button>
      {showShortBarTitle && <Title.Small>夏目天子的博客</Title.Small>}
    </div>
    <div className={styles.content}>
      <div>
        <Title.Small>兴趣使然的博客</Title.Small>
      </div>
      <div className={styles.subTitle}>
        <Text.Micro>白嫖是我快乐</Text.Micro>
      </div>
    </div>
  </div>
);
