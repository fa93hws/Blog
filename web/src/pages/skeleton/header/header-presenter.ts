import { observable, action } from 'mobx';

export class GlobalHeaderStore {
  @observable.ref
  showShortBarTitle = false;

  @action
  setShowShortBarTitle(val: boolean) {
    this.showShortBarTitle = val;
  }
}
