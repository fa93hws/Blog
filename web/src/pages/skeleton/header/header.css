@value gridBaseline from '~components/metrics.css';
@value colorBlue from '~components/colors.css';
@value monoWhite, monoWhiteA50, monoWhiteA200, monoBlackA250 from '~components/colors.css';
@value colorPurpleLight from '~components/colors.css';

@value shortHeaderHeight: calc(gridBaseline * 7);

.iconButton {
  margin-left: calc(gridBaseline * 2);
}

.icon {
  fill: monoWhite;
}

.container {
  background-color: colorBlue;
  color: monoWhiteA50;
  height: calc(gridBaseline * 30);
}

.shortBar {
  position: fixed;
  top: 0;
  height: shortHeaderHeight;
  width: 100%;
  display: flex;
  align-items: center;
  background-color: colorBlue;
}

.withBottomShadow {
  box-shadow: 0 0 gridBaseline 0 monoBlackA250;
}

.content {
  padding-left: calc(gridBaseline * 30);
  padding-top: calc(gridBaseline * 13);
}

.subTitle {
  color: colorPurpleLight;
}
