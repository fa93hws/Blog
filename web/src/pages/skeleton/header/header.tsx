import * as React from 'react';
import * as classnames from 'classnames';
import { Title, Text } from 'components/typography/typography';
import { Cross, Menu } from 'components/icons/icons';
import { IconButton } from 'components/buttons/icon-button';
import styles from './header.css';

const IconButtonImpl = React.memo(
  ({ type, onClick }: { type: 'cross' | 'menu'; onClick(): void }) => {
    const Icon =
      type === 'cross'
        ? () => <Cross size="small" />
        : () => <Menu size="small" />;
    return (
      <div className={styles.icon}>
        <IconButton size="small" onClick={onClick} Icon={Icon} circle />
      </div>
    );
  },
);

export const Header = React.memo(
  ({
    showShortBarTitle,
    onIconClicked,
    titleRef,
    icon,
  }: {
    showShortBarTitle: boolean;
    onIconClicked(): void;
    titleRef?: React.Ref<HTMLDivElement>;
    icon: 'cross' | 'menu';
  }) => (
    <div className={styles.container}>
      <div
        className={classnames(styles.shortBar, {
          [styles.withBottomShadow]: showShortBarTitle,
        })}
      >
        <IconButtonImpl type={icon} onClick={onIconClicked} />
        {showShortBarTitle && <Title.Small>夏目天子的博客</Title.Small>}
      </div>
      <div className={styles.content}>
        <div ref={titleRef}>
          <Title.Small>兴趣使然的博客</Title.Small>
        </div>
        <div className={styles.subTitle}>
          <Text.Micro>白嫖是我快乐</Text.Micro>
        </div>
      </div>
    </div>
  ),
);
