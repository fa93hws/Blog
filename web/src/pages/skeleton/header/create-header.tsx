import * as React from 'react';
import { observer } from 'mobx-react';
import { Header } from './header';
import { GlobalHeaderStore } from './header-presenter';
import headerStyles from './header.css';

const THRESHOLD = 0.9;

export function createHeader({
  onCloseClicked,
  onMenuClicked,
}: {
  onCloseClicked(): void;
  onMenuClicked(): void;
}) {
  const presenter = new GlobalHeaderStore();
  const intersectionObserver = new IntersectionObserver(
    entries => {
      presenter.setShowShortBarTitle(entries[0].intersectionRatio < THRESHOLD);
    },
    {
      // TODO Fix types
      rootMargin: `-${(headerStyles as any).shortHeaderHeight}`,
      threshold: THRESHOLD,
    },
  );
  const titleRef = React.createRef<HTMLDivElement>();
  const GlobalHeader = ({ icon }: { icon: 'cross' | 'menu' }) => {
    React.useEffect(() => {
      if (titleRef.current == null) {
        throw new Error('ref to title should not be null');
      }
      intersectionObserver.observe(titleRef.current);
    }, []);
    const onIconClicked = icon === 'cross' ? onCloseClicked : onMenuClicked;
    return (
      <Header
        showShortBarTitle={presenter.showShortBarTitle}
        onIconClicked={onIconClicked}
        titleRef={titleRef}
        icon={icon}
      />
    );
  };
  return observer(GlobalHeader);
}
