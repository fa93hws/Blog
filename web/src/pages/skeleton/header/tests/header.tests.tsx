import * as React from 'react';
import { Header } from '../header';

describe('Header', () => {
  it('renders the header without top bar title and icon is cross', () => {
    expect(
      <Header
        showShortBarTitle={false}
        onCloseClicked={jest.fn()}
        onMenuClicked={jest.fn()}
        icon="cross"
      />,
    ).toMatchRenderedSnapshot();
  });

  it('renders the header with top bar title and icon is cross', () => {
    expect(
      <Header
        showShortBarTitle={true}
        onCloseClicked={jest.fn()}
        onMenuClicked={jest.fn()}
        icon="cross"
      />,
    ).toMatchRenderedSnapshot();
  });

  it('renders the header without top bar title and icon is menu', () => {
    expect(
      <Header
        showShortBarTitle={false}
        onCloseClicked={jest.fn()}
        onMenuClicked={jest.fn()}
        icon="menu"
      />,
    ).toMatchRenderedSnapshot();
  });

  it('renders the header with top bar title and icon is menu', () => {
    expect(
      <Header
        showShortBarTitle={true}
        onCloseClicked={jest.fn()}
        onMenuClicked={jest.fn()}
        icon="menu"
      />,
    ).toMatchRenderedSnapshot();
  });
});
