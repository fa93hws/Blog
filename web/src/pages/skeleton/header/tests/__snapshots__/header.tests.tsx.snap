// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Header renders the header with top bar title and icon is cross 1`] = `
<div
  class="container"
>
  <div
    class="shortBar withBottomShadow"
  >
    <div
      class="icon"
    >
      <button
        class="iconButton small circle"
      >
        <span
          aria-hidden="true"
          class="iconSpan"
        >
          <svg
            height="16"
            version="1.1"
            viewBox="0 0 16 16"
            width="16"
            xmlns="http://www.w3.org/2000/svg"
          >
            <g
              transform="scale(1)"
            >
              src/components/icons/assets/cross.svg
            </g>
          </svg>
        </span>
      </button>
    </div>
    <h4
      class="titleSmall"
    >
      夏目天子的博客
    </h4>
  </div>
  <div
    class="content"
  >
    <div>
      <h4
        class="titleSmall"
      >
        兴趣使然的博客
      </h4>
    </div>
    <div
      class="subTitle"
    >
      <p
        class="textMicro"
      >
        白嫖是我快乐
      </p>
    </div>
  </div>
</div>
`;

exports[`Header renders the header with top bar title and icon is menu 1`] = `
<div
  class="container"
>
  <div
    class="shortBar withBottomShadow"
  >
    <div
      class="icon"
    >
      <button
        class="iconButton small circle"
      >
        <span
          aria-hidden="true"
          class="iconSpan"
        >
          <svg
            height="16"
            version="1.1"
            viewBox="0 0 16 16"
            width="16"
            xmlns="http://www.w3.org/2000/svg"
          >
            <g
              transform="scale(1)"
            >
              src/components/icons/assets/menu.svg
            </g>
          </svg>
        </span>
      </button>
    </div>
    <h4
      class="titleSmall"
    >
      夏目天子的博客
    </h4>
  </div>
  <div
    class="content"
  >
    <div>
      <h4
        class="titleSmall"
      >
        兴趣使然的博客
      </h4>
    </div>
    <div
      class="subTitle"
    >
      <p
        class="textMicro"
      >
        白嫖是我快乐
      </p>
    </div>
  </div>
</div>
`;

exports[`Header renders the header without top bar title and icon is cross 1`] = `
<div
  class="container"
>
  <div
    class="shortBar"
  >
    <div
      class="icon"
    >
      <button
        class="iconButton small circle"
      >
        <span
          aria-hidden="true"
          class="iconSpan"
        >
          <svg
            height="16"
            version="1.1"
            viewBox="0 0 16 16"
            width="16"
            xmlns="http://www.w3.org/2000/svg"
          >
            <g
              transform="scale(1)"
            >
              src/components/icons/assets/cross.svg
            </g>
          </svg>
        </span>
      </button>
    </div>
  </div>
  <div
    class="content"
  >
    <div>
      <h4
        class="titleSmall"
      >
        兴趣使然的博客
      </h4>
    </div>
    <div
      class="subTitle"
    >
      <p
        class="textMicro"
      >
        白嫖是我快乐
      </p>
    </div>
  </div>
</div>
`;

exports[`Header renders the header without top bar title and icon is menu 1`] = `
<div
  class="container"
>
  <div
    class="shortBar"
  >
    <div
      class="icon"
    >
      <button
        class="iconButton small circle"
      >
        <span
          aria-hidden="true"
          class="iconSpan"
        >
          <svg
            height="16"
            version="1.1"
            viewBox="0 0 16 16"
            width="16"
            xmlns="http://www.w3.org/2000/svg"
          >
            <g
              transform="scale(1)"
            >
              src/components/icons/assets/menu.svg
            </g>
          </svg>
        </span>
      </button>
    </div>
  </div>
  <div
    class="content"
  >
    <div>
      <h4
        class="titleSmall"
      >
        兴趣使然的博客
      </h4>
    </div>
    <div
      class="subTitle"
    >
      <p
        class="textMicro"
      >
        白嫖是我快乐
      </p>
    </div>
  </div>
</div>
`;
